### PR TITLE
Epic 8459: Show data availability for NI data

### PR DESCRIPTION
**Pull request by labkey-bpatel**
#### Rationale
Add data availability indicators for the Non-integrated data section on the Study page, and under Learn About > Studies. Additionally, remove data availability headers from under 'Integrated Data', and add that info to the tooltip instead.
Spec [here](https://docs.google.com/document/d/1O8SX0BHReEWdNHSLmBZHVIuDv95A0UJts1oBe5lJZE4/edit#).

#### Changes
* Remove data availability header from under Integrated Data section.
* Add data availability indicators for Non-integrated Data on the Study page.
* Add assay and publication counts to Learn About/Studies page under Data Added column along.
* Use valid downloadable link as an "available" indicator for NI data.
* Update tooltips for Integrated and Non-integrated data.

### CODE REVIEW

File: webapp/Connector/src/app/store/Study.js [hunk 1]
```diff
@@ -283,6 +326,9 @@ Ext.define('Connector.app.store.Study', {
                             study.cavd_affiliation_file_has_permission = this.documentData[d].accessible;
                             study.cavd_affiliation_file_path = Connector.plugin.DocumentValidation.getStudyDocumentUrl(this.documentData[d].filename, study.study_name, this.documentData[d].document_id);
                         }
+                        if (this.documentData[d].document_type === 'Non-Integrated Assay') {
+
```
Comment (XingY) [inline webapp/Connector/src/app/store/Study.js]: empty if?

File: webapp/Connector/src/app/store/Study.js [hunk 2]
```diff
@@ -128,6 +152,25 @@ Ext.define('Connector.app.store.Study', {
...
         this._onLoadComplete();
+
+        //For using valid downloadable link as an "availability" indicator for NI data
+        this.niLinkValidationCount = 0;
+        this.niDocumentData = this.documentData.filter(function (doc) { return doc.document_type === "Non-Integrated Assay" });
+
+        if (this.niDocumentData.length > 0) {
+            for (var itr = 0; itr < this.niDocumentData.length; itr++) {
+
+                var docIsValidAction = function (doc, status) {
```
Comment (XingY) [inline webapp/Connector/src/app/store/Study.js]: Seems this function definition can be moved out of the loop and reused.

File: webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js [hunk 3]
```diff
@@ -68,9 +56,14 @@ Ext.define('Connector.view.module.NonIntegratedDataAvailability', {
...
                             id = Ext.id();
+
+                    var toolTipMsg_available = "Non-integrated data added to Dataspace";
+                    var toolTipMsg_restricted = "Non-integrated data access is restricted";
+                    var toolTipMsg_notAdded = "Non-integrated data has not been added at this time";
+
                     if (record.data.dataStatus && dataLink) {
                         dataLink.on('mouseenter', this.showDataStatusTooltip, this, {
-                            status: record.data.dataStatus,
+                            status: !!record.data.isLinkValid ? (record.data.hasPermission ? toolTipMsg_available : toolTipMsg_restricted) : toolTipMsg_notAdded,
```
Comment (XingY) [inline webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js:78]: I don't think !! is needed here

File: webapp/Connector/src/app/store/Study.js [hunk 4]
```diff
@@ -20,6 +20,30 @@ Ext.define('Connector.app.store.Study', {
...
     },
 
+    validateDocumentLink : function (doc, callback) {
+        if (doc.isLinkValid === undefined) {
+            LABKEY.Ajax.request({
+                method: 'HEAD',
+                url: Connector.plugin.DocumentValidation.getStudyDocumentUrl(doc.filename, doc.prot, doc.document_id),
+                success: LABKEY.Utils.getCallbackWrapper(function (json, response) {
+                    if (200 === response.status) {
+                        callback.call(this, doc, true);
```
Comment (XingY) [inline webapp/Connector/src/app/store/Study.js]: callback.call(this, doc, 200 === response.status);

**Review by XingY — APPROVED** [review]

File: webapp/Connector/src/app/store/Study.js (hunk 1 above)
Comment (labkey-bpatel) [reply webapp/Connector/src/app/store/Study.js]: now removed

File: webapp/Connector/src/app/store/Study.js (hunk 2 above)
Comment (labkey-bpatel) [reply webapp/Connector/src/app/store/Study.js]: yup, now done

**Comment by labkey-bpatel** [comment]
1 unrelated test failure, merging..

**Pull request merged**